kotlin-analysis/src/main/kotlin/org/jetbrains/dokka/analysis/KotlinAnalysis.kt

```diff
@@ -6,19 +6,25 @@ import org.jetbrains.dokka.DokkaConfiguration.DokkaSourceSet
 import org.jetbrains.dokka.DokkaSourceSetID
 import org.jetbrains.dokka.model.SourceSetDependent
 import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.utilities.DokkaLogger
 
-fun KotlinAnalysis(context: DokkaContext): KotlinAnalysis {
-    val environments = context.configuration.sourceSets.associateWith { sourceSet ->
+fun KotlinAnalysis(sourceSets: List<DokkaSourceSet>, logger: DokkaLogger): KotlinAnalysis {
+    val environments = sourceSets.associateWith { sourceSet ->
         createEnvironmentAndFacade(
-            logger = context.logger,
-            configuration = context.configuration,
+            logger = logger,
+            sourceSets = sourceSets,
             sourceSet = sourceSet
         )
     }
 
     return KotlinAnalysisImpl(environments)
 }
 
+@Deprecated(message = "Construct using DokkaConfiguration and logger",
```

Comment:
The message should rather be: "Construct using list of DokkaSourceSets and logger" indicating what has to be used to not contradict the `replaceWith` message